pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -241,7 +241,7 @@ func updateSubnetsConnectivityByTransitGateway(src, dst VPCResourceIntf,
 	c *VPCConfig) (
 	*netset.TransportSet, error) {
 	// assuming a single router representing the tgw for a "MultipleVPCsConfig"
-	if len(c.RoutingResources) != 1 {
+	if len(c.RoutingResources) != 2 { // expecting tgw and sgw (virtual gateway)
```

Comment:
Will this hold for AWS as well?

Reply:
aws does not get here

Reply:
@adisos , I need your help here. Is the change to the check valid?

Reply:
I would change this check.. and also the the next line in which it assumes that `tgw` is at index `0` of this slice. Instead, I think it is better to check here that `c.RoutingResources` contains exactly one router of kind `TGW` for MultipleVPCsConfig.

Reply:
fixed